server/swap/swap.go

```diff
@@ -753,11 +772,18 @@ func (s *Swapper) processInit(msg *msgjson.Message, params *msgjson.Init, stepIn
 	actor.status.swapVout = params.Vout
 	actor.status.swapTime = time.Now()
 	stepInfo.match.Status = stepInfo.nextStep
-	// NOTE: If UpdateMatch isn't non-blocking, this will need to be done
-	// differently.
-	s.storage.UpdateMatch(stepInfo.match.Match)
+	match := stepInfo.match.Match
 	s.matchMtx.Unlock()
 
+	if err = s.storage.UpdateMatch(match); err != nil {
+		log.Errorf("UpdateMatch failed: %v", err)
+		// TODO: Need a way to signal to the market any storage failures.
+		// Perhaps the archivist itself can switch into an error state that the
+		// Market can detect, allowing Market to halt new order processing. In
+		// that case, the UpdateMatch error can be ignored here and the call can
+		// be made non-blocking.
```

Comment:
Haven't you solved this now?

Reply:
Maybe, I just don't know yet.  Will revise the comment.

Reply:
Correction: solved enough for the swapper. Will make this non-blocking.